packages/client/hmi-client/src/services/data.ts

```diff
@@ -365,88 +312,60 @@ const getRelatedDocuments = async (docid: string, dataset: string | null) => {
 };
 
 const searchXDDArticles = async (term: string, xddSearchParam?: XDDSearchParams) => {
-	const limitResultsCount = xddSearchParam?.perPage ?? XDD_RESULT_DEFAULT_PAGE_SIZE;
+	let searchParams = `term=${term}`;
+	const url = '/xdd/documents?';
+
+	if (xddSearchParam) {
+		Object.entries(xddSearchParam).forEach(([key]) => {
+			// Loop through searchParam attributes
+			if (xddSearchParam[key]) {
+				const urlKey = key.replace(/[A-Z]/g, (c) => `_${c.toLowerCase()}`); // Later maybe make attributes have underscore format from the get go?
```

Comment:
One reason I am against this change is because the code
`key.replace(/[A-Z]/g, (c) => `_${c.toLowerCase()}`);
` is not too clear on what exactly it is doing as it is not documented and assumes some knowledge about regex (as opposed to the old code which was more explicit).

Also, please note here how the parameters are assumed be passed to the server with underscore in their naming. While this is currently the case, it shouldn't be because the server is supposed to be receiving proper parameter/variable names without the underscore and without forcing the client side to do this naming transformation. The initial implementation, however, was forced to use underscore because the server is not yet able to handle such naming. I would prefer code refactoring that properly fix this issue rather than converting the multiple "if" statements into a switch statement!